### PR TITLE
Active record basics

### PR DESCRIPTION
**Pull request by Krutik31**
Applied CRUD operations on the Product model in console. 
Added simple validations & callbacks.
Used bootstrap for designing index & show pages.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
-  get 'home/index'
+  get '/product/', to: "product#index"
+  get '/home/', to: "home#index"
+  get 'product/:id', to: "product#show"
```
Comment (purvak-simform) [inline config/routes.rb]: @Krutik31 There is no need to create manual routes for product. You can write it like this
  resources :product, only: [:index, :show]

File: app/views/product/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        <th>Action</th>
+      </tr>
+    </thead>
+    <tbody>
+      <% @products.each do |product| %>
+        <tr>
+          <td><%= product.id %></td>
+          <td><%= product.title %></td>
+          <td>
+            <a href="/product/<%= product.id %>" class="btn btn-sm btn-outline-success">View</a>
```
Comment (purvak-simform) [inline app/views/product/index.html.erb]: @Krutik31 You are creating a link here, so use rails method link_to
Check it here how can a link be created in rails
https://www.rubyguides.com/2019/05/rails-link_to-method/

File: app/views/product/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <strong>Price: </strong>
+    <%= @product.price %>
+  </h5><hr>
+
+  <h5>
+    <strong>Brand: </strong>
+    <%= @product.brand %>
+  </h5><hr>
+  
+  <a class="btn btn-flat btn-danger float-left" href="/product">Back</a>
```
Comment (purvak-simform) [inline app/views/product/show.html.erb]: Same as above

File: config/routes.rb (hunk 1 above)
Comment (Krutik31) [reply config/routes.rb]: Changes made on routes

File: app/models/product.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  after_create :createdmsg
+  def createdmsg
+    puts "Message after create command from 'after_create' callback."
+  end
+
+  after_commit :commitedmsg
+  def commitedmsg
+    puts "Message after successful commiting from 'after_commit' callback."
+  end
```
Comment (purvak-simform) [inline app/models/product.rb:22]: @Krutik31 We don't write callback then method, write all callbacks together and methods downside
And take a caution while keeping methods name it should be meaningful
`commitedmsg` instead change it to something like `display_commit_callback_message`

Similarly change for above

File: config/routes.rb (hunk 1 above)
Comment (purvak-simform) [reply config/routes.rb]: @Krutik31  in routes controller name are always plural

**Review by purvak-simform — APPROVED** [review]

**Pull request merged**